packages/amplify-cli/src/input-manager.ts

```diff
@@ -110,17 +110,41 @@ export function verifyInput(pluginPlatform: PluginPlatform, input: Input): Input
         result.helpCommandAvailable = true;
       }
 
+      // verify if `input.command` is an actual command.
       if (commands && commands!.includes(input.command!)) {
         result.verified = true;
         break;
       }
 
+      // verify if `input.command` is an alias for a command.
       if (commandAliases && Object.keys(commandAliases).includes(input.command!)) {
         input.command = commandAliases[input.command!];
         result.verified = true;
         break;
       }
 
+      // if `input.command` is not a command name or an alias for a command, check the
+      // first sub-command for a verb / noun swap (i.e. `env add` versus `add env`).
+      if (commands && input.subCommands && commands!.includes(input.subCommands[0])) {
+        const command = input.subCommands[0];
+        input.subCommands[0] = input.command!;
+        input.command = command;
+
+        result.verified = true;
+        break;
+      }
+
+      // same as above, but check if the first sub-command is an alias.
+      if (commandAliases && input.subCommands && Object.keys(commandAliases).includes(input.subCommands[0])) {
```

Comment:
Instead of `Object.keys(commandAliases).includes(input.subCommands[0])`, I think you should be able to do a `hasOwnProperty()` check to accomplish the same thing.

Reply:
For sure! I'll do this now.